lib/animina_web/live/chat_live.ex

```diff
@@ -416,6 +402,68 @@ defmodule AniminaWeb.ChatLive do
     {:noreply, socket}
   end
 
+  @impl true
+  def handle_info({request_pid, {:data, %{"done" => false, "response" => chunk}}}, socket) do
+    socket =
+      case socket.assigns.current_request do
+        %{pid: ^request_pid} ->
+          generated_message = socket.assigns.generated_message <> chunk
+
+          socket
+          |> assign(:generated_message, generated_message)
+          |> assign(:suggested_messages, ChatCompletion.parse_message(generated_message))
+
+        _ ->
+          socket
+      end
+
+    {:noreply, socket}
+  end
+
+  @impl true
+  def handle_info({request_pid, {:data, %{"done" => true, "response" => response}}}, socket) do
+    socket =
+      case socket.assigns.current_request do
+        %{pid: ^request_pid} ->
+          generated_message = socket.assigns.generated_message <> response
+
+          socket
+          |> assign(:generated_message, generated_message)
+          |> assign(:current_request, nil)
+          |> assign(:show_use_ai_button, true)
+          |> assign(:generated_message, false)
+
+        _ ->
+          socket
+      end
+
+    {:noreply, socket}
+  end
+
+  @impl true
+  def handle_info({:generate_messages, _}, socket) do
+    socket =
+      case ChatCompletion.request_message(socket.assigns.sender, socket.assigns.receiver) do
+        {:ok, task} ->
+          socket
+          |> assign(:current_request, task)
+          |> assign(:generated_message, "")
+          |> assign(:show_use_ai_button, true)
+
+        {:error, _} ->
+          socket
+          |> put_flash(:error, "Could not generate message with AI, Kindly Try Again")
```

Comment:
Always use `gettext/1` for any kind of message.

Reply:
Thanks I'll always use it